sequencer/src/bin/deploy.rs

```diff
@@ -245,15 +256,21 @@ async fn main() -> anyhow::Result<()> {
             let l1 = l1.clone();
             let orchestrator_url = opt.orchestrator_url.clone();
             async move {
-                let light_client = LightClient::new(
+                let light_client = LightClientMock::new(
                     contracts
                         .deploy_fn(Contract::LightClient, |contracts| {
-                            deploy_light_client_contract(l1.clone(), contracts).boxed()
+                            deploy_light_client_contract(
+                                l1.clone(),
+                                contracts,
+                                opt.use_mock_contract,
+                            )
+                            .boxed()
                         })
```

Comment:
wait, shouldn't you gate this using `use_mock_contract: bool` ?
this code seems to always use `LightClientMock` in the `main()`?